File: app/src/main/java/com/tt/handsomeman/viewmodel/SignUpViewModel.java
package com.tt.handsomeman.viewmodel;

import android.util.Patterns;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.tt.handsomeman.R;
import com.tt.handsomeman.model.SignUpFormState;

public class SignUpViewModel extends ViewModel {

    private MutableLiveData<SignUpFormState> signUpFormState = new MutableLiveData<>();

    public LiveData<SignUpFormState> getSignUpFormState() {
        return signUpFormState;
    }

    public void signUpDateChanged(String name, String mail, String password, String rePassword) {
        if (!(isNameValid(name))) {
            signUpFormState.setValue(new SignUpFormState(R.string.not_valid_name, null, null, null));
        }
        if (!(isMailValid(mail))) {
            signUpFormState.setValue(new SignUpFormState(null, R.string.not_valid_mail, null, null));
        }
        if (!(isPasswordValid(password))) {
            signUpFormState.setValue(new SignUpFormState(null, null, R.string.not_valid_password, null));
        }
        if (!(isMatchPassword(password, rePassword))) {
            signUpFormState.setValue(new SignUpFormState(null, null, null, R.string.password_not_match));
        }
        if (isNameValid(name) && isMailValid(mail) && isPasswordValid(password) && isMatchPassword(password, rePassword)) {
            signUpFormState.setValue(new SignUpFormState(true));
        }
    }

    private boolean isNameValid(String name) {
        return name != null && name.trim().length() > 4;
    }

    private boolean isMailValid(String mail) {
        if (mail == null) {
            return false;
        }
        if (!(Patterns.EMAIL_ADDRESS.matcher(mail).matches())) {
            return false;
        } else {
            return !mail.trim().isEmpty();
        }
    }

    private boolean isPasswordValid(String password) {
        return password != null && password.trim().length() > 5;
    }

    private boolean isMatchPassword(String password, String rePassword) {
        if (rePassword != null && !(rePassword.equals(password))) {
            return false;
        }
        return true;
    }
}

File: app/src/main/java/com/tt/handsomeman/viewmodel/SignUpViewModel.kt
package com.tt.handsomeman.viewmodel

import android.util.Patterns

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import com.tt.handsomeman.R
import com.tt.handsomeman.model.SignUpFormState

class SignUpViewModel : ViewModel() {

    private val signUpFormState = MutableLiveData<SignUpFormState>()

    fun getSignUpFormState(): LiveData<SignUpFormState> {
        return signUpFormState
    }

    fun signUpDateChanged(name: String, mail: String, password: String, rePassword: String) {
        if (!isNameValid(name)) {
            signUpFormState.value = SignUpFormState(R.string.not_valid_name, null, null, null)
        }
        if (!isMailValid(mail)) {
            signUpFormState.value = SignUpFormState(null, R.string.not_valid_mail, null, null)
        }
        if (!isPasswordValid(password)) {
            signUpFormState.value = SignUpFormState(null, null, R.string.not_valid_password, null)
        }
        if (!isMatchPassword(password, rePassword)) {
            signUpFormState.value = SignUpFormState(null, null, null, R.string.password_not_match)
        }
        if (isNameValid(name) && isMailValid(mail) && isPasswordValid(password) && isMatchPassword(password, rePassword)) {
            signUpFormState.value = SignUpFormState(true)
        }
    }

    private fun isNameValid(name: String?): Boolean {
        return name != null && name.trim { it <= ' ' }.length > 4
    }

    private fun isMailValid(mail: String?): Boolean {
        if (mail == null) {
            return false
        }
        return if (!Patterns.EMAIL_ADDRESS.matcher(mail).matches()) {
            false
        } else {
            !mail.trim { it <= ' ' }.isEmpty()
        }
    }

    private fun isPasswordValid(password: String?): Boolean {
        return password != null && password.trim { it <= ' ' }.length > 5
    }

    private fun isMatchPassword(password: String, rePassword: String?): Boolean {
        return if (rePassword != null && rePassword != password) {
            false
        } else true
    }
}
